ffi/src/scan.rs

```diff
@@ -275,40 +263,38 @@ pub unsafe extern "C" fn get_from_map(
     map: &mut CStringMap,
     key: KernelStringSlice,
     allocate_fn: AllocateStringFn,
-) -> *mut c_void {
+) -> NullableCvoid {
     let string_key = String::try_from_slice(key);
-    match map.values.get(&string_key) {
-        Some(v) => allocate_fn(v.as_str().into()),
-        None => std::ptr::null_mut(),
-    }
+    map.values
+        .get(&string_key)
+        .and_then(|v| allocate_fn(v.as_str().into()))
```

Comment:
rescuing https://github.com/delta-incubator/delta-kernel-rs/pull/175#discussion_r1581352849
>  more and more, I lean toward implementing the above [`impl<T: AsRef<[u8]>> From<T> for KernelStringSlice`], which would allow simply:

```suggestion
        .and_then(|v| allocate_fn(v.into()))
```